Sources/TuistGraph/Models/Dependencies/SwiftPackageManagerDependencies.swift

```diff
@@ -5,19 +5,39 @@ public struct SwiftPackageManagerDependencies: Equatable {
     /// List of packages that will be installed using Swift Package Manager.
     public let packages: [Package]
 
+    /// List of options for Carthage installation.
+    public let options: Set<Options>
+
     /// Initializes a new `SwiftPackageManagerDependencies` instance.
-    /// - Parameter packages: List of packages that will be installed using Swift Package Manager.
-    public init(_ packages: [Package]) {
+    /// - Parameters:
+    ///    - packages: List of packages that will be installed using Swift Package Manager.
+    ///    - options: List of options for Swift Package Manager installation.
+    public init(
+        _ packages: [Package],
+        options: Set<Options>
+    ) {
         self.packages = packages
+        self.options = options
+    }
+}
+
+extension SwiftPackageManagerDependencies {
+    /// Returns the version of swift tools that should be used to resolve dependencies or `nil` when the version is unspecified.
+    public var swiftToolsVersion: String? {
+        options
+            .compactMap { option -> String? in
+                switch option {
+                case let .swiftToolsVersion(version):
+                    return version
+                }
+            }.first
     }
 
     /// Returns `Package.swift` representation.
     ///
     /// **NOTE** It is a temporary solution until Apple resolves: https://forums.swift.org/t/pitch-package-editor-commands/42224
```

Comment:
Unrelated: do we have a ticket to track this?

Reply:
no, I don't have

Reply:
> Wondering what happens if users specify an invalid tools version?

Probably `swift package [command]` command will fail and Tuist will forward error message to a user.

Reply:
I made a quick test.

I set invalid `swift-tools-version`:

```swift
import ProjectDescription

let dependencies = Dependencies(
    swiftPackageManager: .swiftPackageManager(
        [
            .package(url: "https://github.com/SnapKit/SnapKit.git", .upToNextMajor(from: "5.0.1"))
        ],
        options: [
            .swiftToolsVersion("10.4.0"),
        ]
    ),
    platforms: [.iOS]
)
```

and I got:

<img width="1221" alt="Zrzut ekranu 2021-05-23 o 20 55 17" src="https://user-images.githubusercontent.com/4774319/119273046-5fa04a80-bc09-11eb-935c-ab1582fe01d8.png">

Reply:
> Unrelated: do we have a ticket to track this?

Done ;)